java-generic-mqtt/src/main/java/com/cumulocity/generic/mqtt/client/model/GenericMqttMetadata.java

```diff
@@ -0,0 +1,66 @@
+package com.cumulocity.generic.mqtt.client.model;
+
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.Map;
+
+/**
+ * Metadata for {@link GenericMqttMessage}.
+ */
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class GenericMqttMetadata {
+
+    /**
+     * The client id of the MQTT client.
+     */
+    private String clientId;
+
+    /**
```

Comment:
nitpick: Id prefer 'The message's id'. 
question: Is it unique, per subscription, per client? Do we know? I'd guess it's unique to a pulsar topic. I'd also assume it will rollover at max value. Does it rollver to negative values or zero?

Sorry for awkward questions - I don't know the answers to these myself. But users might want to know if there are constraints in how they use the Id.

Reply:
That id goes directly from mqtt broker and as far as I recall it is unique per client id connection, once client reconnects counter is started from the beginning

Reply:
Thanks. It's a bit of a nitpick, maybe we should say it's session scoped, resetting to zero on reconnection